src/support.sch

```diff
@@ -303,6 +304,30 @@
         </rule>
     </pattern>
 
+    <pattern id="ext-link">
+        <rule context="ext-link[not(contains(@xlink:href, ':') or ancestor-or-self::*[contains(@xml:base, ':')])]" role="warn">
```

Comment:
Not correct, but think it's the best we can do with XPath 1...

Reply:
XPath 1 has a function `starts-with($variable,$test)`. We could use this to mandate that only links that begin with a known uri scheme are supported, example:
```xml
<rule context="ext-link" role="warn">
            <assert test="starts-with(@xlink:href,'https://') or starts-with(@xlink:href,'http://') or starts-with(@xlink:href,'ftp://') or starts-with(@xlink:href,'mailto:') or starts-with(@xlink:href,'tel:') or starts-with(@xlink:href,'git://')">
                &lt;<name/>&gt; without an absolute @xlink:href is ignored.
            </assert>
        </rule>
```

Presumably there are hundreds of valid uri schemes, but if we were to output a message for the most used uri schemes, which again I assume is a handful, then we could cover the majority of cases.

Reply:
I'll leave for now, something we can maybe revisit (don't want to block potential but obscure schemes).